Code/src/main/java/nl/utwente/viskell/ui/components/ConnectionAnchor.java

```diff
@@ -114,7 +26,14 @@ public void handle(InputEvent event) {
      */
     public ConnectionAnchor(Block block) {
         this.block = block;
-        this.new AnchorHandler(block.getPane().getConnectionCreationManager());
+        
+        this.addEventHandler(MouseEvent.MOUSE_PRESSED, this::handleMousePress);
+        this.addEventHandler(MouseEvent.MOUSE_DRAGGED, this::handleMouseDrag);
+        this.addEventHandler(MouseEvent.MOUSE_RELEASED, this::handleMouseRelease);
+        
+        this.addEventHandler(TouchEvent.TOUCH_PRESSED, this::handleTouchPress);
+        this.addEventHandler(TouchEvent.TOUCH_MOVED, this::handleTouchMove);
+        this.addEventHandler(TouchEvent.TOUCH_RELEASED, this::handleTouchRelease);
```

Comment:
This might be an opportunity to make the names for these more consistent: this uses handleFoo, other bits use onFoo.